content/en/security/application_security/enabling/compatibility/nodejs.md

```diff
@@ -128,24 +125,26 @@ The following operating systems are officially supported by `dd-trace`. Any oper
 - **Threat Protection** also works at the HTTP request (input) layer, and so works for all databases by default, even those not listed in the table below.
 
 
-| Framework                 | Versions | Threat Detection supported? | Threat Protection supported? | Vulnerability Management for Code-level supported? |
-| ----------------------- | ---------- | --------------- | ---------------------------------------------- | ---------------------------------------------- |
-| [cassandra-driver][28] | `>=3`    |{{< X >}}|{{< X >}} |          |
-| [couchbase][29]        | `^2.4.2` |{{< X >}}|          |          |
-| [elasticsearch][30]    | `>=10`   |{{< X >}}|          |          |
-| [ioredis][31]          | `>=2`    |{{< X >}}|{{< X >}} |          |
-| [knex][32]             | `>=0.8`  |{{< X >}}|{{< X >}} |          |
-| [mariadb][5]           | `>=3`    |{{< X >}}|{{< X >}} |          |
-| [memcached][33]        | `>=2.2`  |{{< X >}}|{{< X >}} |          |
-| [mongodb-core][34]     | `>=2`    |{{< X >}}|{{< X >}} |          |
-| [mysql][35]            | `>=2`    |{{< X >}}|          |{{< X >}} |
-| [mysql2][36]           | `>=1`    |{{< X >}}|{{< X >}} |{{< X >}} |
-| [oracledb][37]         | `>=5`    |{{< X >}}|{{< X >}} |          |
-| [pg][38]               | `>=4`    |{{< X >}}|{{< X >}} |{{< X >}} |
-| [redis][39]            | `>=0.12` |{{< X >}}|{{< X >}} |          |
-| [sharedb][40]          | `>=1`    |{{< X >}}|{{< X >}} |          |
-| [tedious][41]          | `>=1`    |{{< X >}}|{{< X >}} |          |
-| [sequelize][42]        | `>=4`    |         |          | {{< X >}}|
+| Framework                | Versions  | Threat Detection supported? | Threat Protection supported? | Vulnerability Management for Code-level supported? |
```

Comment:
is this table accurate ? do we support everything said here ??

Reply:
Most of then are to connect to databases, as the waf is running in request start/end and not in the libraries, the waf job is not related with them, so we can detect threats if they are in rules in any of them.

The "Vulnerability Management for Code-level supported?" is different, as we need each point instrumentation to detect vulnerabilities, and it is accurate, yes.

But now that I reviewed the table... I don't know why couchbase and elasticsearch have the check in detection and not in protection...
In going to add the `{{< X >}}` in those.

Reply:
For the "Vulnerability Management for Code-level supported?" we need each point instrumentation to detect vulnerabilities, and yes, it is accurate

For Threat columns, most of then don't have specific instrumentations for appsec, we are not doing anything in library to detect them specifically. So our rules should work for any of them. As they are supported in apm, the customer can see that the threat is detected in a trace with database span, and backend can do more things with these data.

But now that I reviewed the table... I don't know why couchbase and elasticsearch have the check in detection and not in protection... I am going to add the {{< X >}} in those.

Reply:
Yeah my point here is do we have any feature related to those ? it feels wrong to advertise we support databases libraries when we don't even test for them in an appsec context. Also why is there even a version start for the support ?? hahaha what version is this referencing if we don't even have instrumentation for it ?

Reply:
> do we have any feature related to those ?

With the frameworks that I added here, yes we have features related.
With the libraries that were here previously, probably yes in the backend waf. I didn't make this list and I don't see myself with enough knowledge of the backend to update this. IIRC @sourabhkatti got this list from apm page.

> Also why is there even a version start for the support ??
Table copied from APM, I guess that we only need the traces.

Reply:
Let's get quick confirmation that this is not just pulled out of thin air, and then LGTM, i'll approve now so you can merge faster